tests/end_to_end/test_import.py

```diff
@@ -91,83 +91,111 @@ def test_import_star(execute):
-            == """import lineapy.utils.__no_imported_submodule_prime
-second_is_prime = lineapy.utils.__no_imported_submodule_prime.is_prime
-"""
-        )
+# import x
+def test_basic_import(execute):
+    code = """import math
+x = math.sqrt(64)
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"] == 8
+    assert res.artifacts["x"] == code
+
+
+# import x as a
+def test_basic_import_as(execute):
+    code = """import math as m
+x = m.sqrt(64)
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"] == 8
+    assert res.artifacts["x"] == code
+
+
+# from x import y
+def test_basic_from_import(execute):
+    code = """from math import sqrt
+x = sqrt(64)
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"] == 8
+    assert res.artifacts["x"] == code
+
+
+# from x import y as a
+def test_basic_from_import_as(execute):
+    code = """from math import sqrt as sq
+x = sq(64)
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"] == 8
+    assert res.artifacts["x"] == code
+
+
+# from x import *
+def test_basic_from_import_starred(execute):
+    code = """from math import *
+x = sqrt(64)
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"] == 8
+    assert res.artifacts["x"] == code
+
+
+# import x.y.z
+def test_nested_import(execute):
+    code = """import matplotlib.pyplot
+x = matplotlib.pyplot.ylabel('label')
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"].__class__.__name__ == "Text"
+    assert res.artifacts["x"] == code
+
+
+# import x.y.z as a
+def test_nested_import_as(execute):
+    code = """import matplotlib.pyplot as plt
+x = plt.ylabel('label')
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"].__class__.__name__ == "Text"
+    assert res.artifacts["x"] == code
+
+
+# from x.y import z
+def test_nested_from_import(execute):
+    code = """from matplotlib.pyplot import ylabel
+x = ylabel('label')
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"].__class__.__name__ == "Text"
+    assert res.artifacts["x"] == code
+
+
+# from x.y import z as a
+def test_nested_from_import_as(execute):
+    code = """from matplotlib.pyplot import ylabel as yl
+x = yl('label')
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"].__class__.__name__ == "Text"
+    assert res.artifacts["x"] == code
+
+
+# from x.y import *
+def test_nested_from_import_starred(execute):
+    code = """from matplotlib.pyplot import *
+x = ylabel('label')
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"].__class__.__name__ == "Text"
+    assert res.artifacts["x"] == code
+
+
+# from x.y import *
+def test_nested_from_import_submodule(execute):
+    code = """from matplotlib import pyplot
+x = pyplot.ylabel('label')
+"""
+    res = execute(code, artifacts=["x"])
+    assert res.values["x"].__class__.__name__ == "Text"
```

Comment:
What is this assertion for?

Reply:
I think this was done because adding a ylabel does _not_ create a straightforward "value" that can be verified. This probably is a roundabout way to ensure that an object of Text type is created. @moustafa-a can verify this.